python/lsst/daf/butler/direct_butler/_direct_butler.py

```diff
@@ -1933,28 +1902,30 @@ def _extract_dimension_records_from_data_ids(
 
         return dimension_records
 
-    def transfer_from(
+    def _prepare_for_import_refs(
         self,
         source_butler: LimitedButler,
         source_refs: Iterable[DatasetRef],
-        transfer: str = "auto",
         skip_missing: bool = True,
         register_dataset_types: bool = False,
         transfer_dimensions: bool = False,
         dry_run: bool = False,
-    ) -> collections.abc.Collection[DatasetRef]:
+    ) -> tuple[
+        defaultdict[_RefGroup, list[DatasetRef]],
+        dict[DimensionElement, dict[DataCoordinate, DimensionRecord]],
+        dict[ResourcePath, bool],
+    ]:
```

Comment:
This feels somewhat uncomprehensible, could a separate `dataclass` describe it better?

Reply:
I was wondering whether you would comment on that. I thought I would get away with it given that it's an internal API... 😉